keps/sig-network/2433-topology-aware-hints/README.md

```diff
@@ -369,12 +376,27 @@ of the following scenarios:
 2. A new Node results in a Service that is able to achieve an endpoint
    distribution below 20% for the first time.
 
+### SameZone Heuristic
+The simplest of all heuristics, this will simply copy the zone of an endpoint to
+the hint of the same endpoint. If there are not endpoints available within a zone,
+that will mean no endpoints will have hints for that zone. As described above,
+when Kube-Proxy cannot find any endpoints with hints for its zone, it will route
+traffic ~evenly across all endpoints, regardless of hints.
+
+### Additional Heuristics
+To enable additional heuristics to be added in the future, we will:
+
+1. Remove the requirement in kube-proxy that the hints annotation must be set to
+   a known value on the associated Service before the values of EndpointSlice
+   hints will be considered.
```

Comment:
Why is this necessary ? 
I assume that every heuristic we may add in the future will probably require changes in kube proxy to implement a different logic for that heuristic. 
I might miss something but is it the case where we just want to avoid maintaining a switch case on the annotation value?

Reply:
Most heuristics should not need proxy changes - if they do, we're doing something wrong.